libs/libc/machine/arch_atomic.c

```diff
@@ -401,3 +425,35 @@ FETCH_XOR(4, uint32_t)
  ****************************************************************************/
 
 FETCH_XOR(8, uint64_t)
+
+/* Clang define the __sync builtins, add #ifndef to avoid
+ * redefined/redeclared problem.
+ */
+
+#ifndef __clang__
+
+/****************************************************************************
+ * Name: __sync_add_and_fetch_1
+ ****************************************************************************/
+
+SYNC_ADD_FETCH(1, uint8_t)
```

Comment:
do we need `__sync_sub_and_fetch` and others?

Reply:
Yes, i will do this in this PR later today.

Reply:
@pkarashchenko Done